lib/tasks/uploads.rake

```diff
@@ -840,8 +833,20 @@ def determine_upload_security_and_posts_to_rebake(uploads_to_update, mark_secure
   uploads_to_adjust_acl_for = []
   posts_to_rebake = {}
 
+  # we do this to avoid a heavier post query, and to make sure we only
+  # get unique posts AND include deleted posts (unscoped)
+  unique_access_control_posts = {}
+  Post.unscoped.select(:id, :topic_id)
+    .includes(topic: :category)
+    .where(id: uploads_to_update.pluck(:access_control_post_id).uniq).find_each do |post|
+    unique_access_control_posts[post.id] = post
```

Comment:
Does this not pull a huge number of posts into memory? Why not just keep the ids and then instead just update `access_control_post_id = id` ?

Reply:
@SamSaffron because the post's topic and that topic's category are used in `UploadSecurity` to determine whether the post has secure media. See:

https://github.com/discourse/discourse/blob/master/lib/upload_security.rb#L51-L53
https://github.com/discourse/discourse/blob/master/app/models/post.rb#L500-L504

Reply:
In the forum I have tested on this results in ~3400 posts in memory for ~93k uploads.